src/seedu/addressbook/data/AddressBook.java

```diff
@@ -38,6 +38,35 @@ public void addPerson(Person toAdd) throws DuplicatePersonException {
         allPersons.add(toAdd);
     }
 
+    /**
+     * Adds a person to the address book.
+     * This method is mean to be used with {@code editPerson(Person toEdit)}.
+     * It silences {@code DuplicatePersonException} since this method is
+     * called after {@code PersonNotFoundException} is thrown.
+     * Both exceptions are mutually exclusive.
+     *
+     */
+    private void addPersonSafe(Person toAdd) {
+        try {
+            addPerson(toAdd);
+        } catch(DuplicatePersonException dpe) {
+            // Will not be run
```

Comment:
I think it is not safe to assume that just because the exception will not be raised now means that no changes to the project might every raise the exception. I think it should be handled well still.